src/net.h

```diff
@@ -1198,6 +1197,16 @@ class CConnman
     int GetExtraFullOutboundCount() const;
     // Count the number of block-relay-only peers we have over our limit.
     int GetExtraBlockRelayCount() const;
+    /**
+     * If we are at capacity for inbound tx-relay peers, attempt to evict one.
+     * @param[in]   protect_peer    NodeId of a peer we want to protect
+     * @return      bool              Returns true if successful (either there is
+     *                                no need for eviction, or a peer was evicted).
+     *                                Returns false, if we are full but couldn't find
+     *                                a peer to evict (all eligible peers are protected)
+     *                                so that the caller can deal with this.
+     */
+    bool EvictTxPeerIfFull(std::optional<NodeId> protect_peer);
```

Comment:
changed to use default values.

Not sure about the rename suggestion, I think I prefer `EvictTxPeerIfFull`, because I think for naming describing what the function does is more important than describing the return value. Otherwise other spots might call this function in the future with the intention to just perform a check, and then might be surprised about the side effects.